saleor/product/utils.py

```diff
@@ -97,32 +99,61 @@ def products_for_cart(user):
     return products
 
 
-def get_variant_picker_data(variants, attributes):
-    data = {'attributes': [], 'variants': []}
-    for attribute in attributes:
-        data['attributes'].append({
+def get_variant_picker_data(product, discounts=None):
+    availability = get_availability(product, discounts)
+    variants = product.variants.all()
+    data = {'productAttributes': [], 'variantAttributes': [], 'variants': []}
```

Comment:
Variant picker renders both the price (which may depend on variant) and variant attributes. It was easier to put product attributes into the variant picker in order to render this whole part of UI properly:

![image](https://cloud.githubusercontent.com/assets/5421321/21850378/96f2b578-d80a-11e6-8dac-ac27587f82ce.png)